BUILD

```diff
@@ -86,10 +86,12 @@ cc_binary(
         "grpc/src/compiler/schema_interface.h",
         "src/flatc_main.cpp",
         "src/idl_gen_cpp.cpp",
+        "src/idl_gen_csharp.cpp",
         "src/idl_gen_dart.cpp",
         "src/idl_gen_general.cpp",
```

Comment:
We're not removing `general` yet?

Reply:
The `general` file still has a few methods `GeneralMakeRule`, `GenerateBinary` that I didn't know how they were used so I left them as is. I would suggest a follow up PR to remove `general` if needed.

Reply:
I'd say we could remove these, possibly move any shared functionality between code generators to `code_generators.cpp`